server.js

```diff
@@ -1,30 +1,192 @@
+import MarkdownIt from 'markdown-it';
+
+// Defines the port the app will run on. Defaults to 8080.
 const port = process.env.PORT || 8080;
-const app = express();
 
-// Add middlewares to enable cors and json body parsing
+// Import a module for listing all endpoints in the app
+const app = require('express')();
+const md = new MarkdownIt();
+
+// Add middlewares to enable cors and json body parsing 
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+
+// Default route to list all available endpoints in the app
+app.get('/', (req, res) => {
+    // MarkdownIt is a library for converting Markdown to HTML.
+    // We use it here to render our API documentation in a more readable format.
+  const documentation = `
+  ## Endpoints
+  
+  ### 1. Get All Endpoints
+  
+  #### Request
+  
+  - **Method:** GET
+  - **Path:** /
+  
+  #### Response
+  
+  - **Status Code:** 200 OK
+  - **Body:** A JSON object listing all available endpoints in the application.
+  
+  ### 2. Get All Netflix Titles
+  
+  #### Request
+  
+  - **Method:** GET
+  - **Path:** /titles
+  
+  #### Response
+  
+  - **Status Code:** 200 OK
+  - **Body:** A JSON array containing all Netflix titles.
+  
+  ### 3. Get a Specific Title by ID
+  
+  #### Request
+  
+  - **Method:** GET
+  - **Path:** /titles/:id
+  
+  #### Parameters
+  
+  - id (integer): The unique identifier of the title.
+  
+  #### Response
+  
+  - **Status Code:** 200 OK if the title is found, 404 Not Found otherwise.
+  - **Body:** A JSON object representing the details of the specific title.
+  
+  ### 4. Get Titles in a Specific Category
+  
+  #### Request
+  
+  - **Method:** GET
+  - **Path:** /titles/categories/:category
+  
+  #### Parameters
+  
+  - category (string): The category for which titles are requested.
+  
+  #### Response
+  
+  - **Status Code:** 200 OK if the category is found, 404 Not Found otherwise.
+  - **Body:** A JSON object containing an array of titles in the specified category.
+  
+  ### 5. Get Titles of a Specific Type (Movie or TV Show)
+  
+  #### Request
+  
+  - **Method:** GET
+  - **Path:** /titles/types/:type
+  
+  #### Parameters
+  
+  - type (string): The type of media (movie or TV show) for which titles are requested.
+  
+  #### Response
+  
+  - **Status Code:** 200 OK if the media type is found, 404 Not Found otherwise.
+  - **Body:** A JSON object containing an array of titles of the specified type.
+  
+  ### 6. Get Titles from a Specific Country
+  
+  #### Request
+  
+  - **Method:** GET
+  - **Path:** /titles/countries/:country
+  
+  #### Parameters
+  
+  - country (string): The country for which titles are requested.
+  
+  #### Response
+  
+  - **Status Code:** 200 OK if the country is found, 404 Not Found otherwise.
+  - **Body:** A JSON object containing an array of titles from the specified country.
+  
+  ## Error Responses
```

Comment:
Really appreciate the description of error messages that sometimes can be a intimidating.